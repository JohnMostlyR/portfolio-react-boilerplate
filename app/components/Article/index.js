import React from 'react';
import PropTypes from 'prop-types';

import StyledArticle from './StyledArticle';

function Article({ loadingContent, ...props }) {
  return <StyledArticle {...props} loadingContent={loadingContent} />;
}

Article.propTypes = {
  loadingContent: PropTypes.bool,
};

Article.defaultProps = {
  loadingContent: false,
};

export default Article;
