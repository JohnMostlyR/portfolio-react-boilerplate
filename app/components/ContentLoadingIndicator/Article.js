import styled from 'styled-components';
import PropTypes from 'prop-types';

const Article = styled.article`
  align-self: center;
  left: 50%;
  position: absolute;
  top: ${({ show }) => (show ? '50%' : '125%')};
  transition: top 0.75s 0s cubic-bezier(0.32, 1.5, 0.29, 1);
  transform: translate3d(-50%, -50%, 0);
  z-index: 99;
`;

Article.propTypes = {
  show: PropTypes.bool,
};

export default Article;
