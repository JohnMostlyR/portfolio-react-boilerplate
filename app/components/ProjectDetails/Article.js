import styled from 'styled-components';
import {
  BASE_FONT_BOLD,
  BASE_LINE_HEIGHT,
  MODULAR_SCALE_FACTOR,
} from '../../styles/typography';

import { theme } from '../../styles/theme';
import { SVGBackgroundImage } from '../../styles/tools';

const {
  inlineLink: { color, textShadowColor },
} = theme.projectDetails;
const H2_LINE_HEIGHT = 1.2;
const EXTERNAL_LINK_ICON = SVGBackgroundImage(`<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512" aria-hidden="true">
    <path
      fill="${color}"
      d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
    />
  </svg>`);

const Article = styled.article`
  max-width: 51em;

  a {
    position: relative;
    background-color: transparent;
    background-image: linear-gradient(currentColor, currentColor);
    background-origin: padding-box;
    background-position: center bottom 10%;
    background-repeat: no-repeat;
    background-size: 100% 1.5px;
    color: ${color};
    cursor: pointer;
    text-decoration: none;
    text-shadow: 3px 0 ${textShadowColor}, 2px 0 ${textShadowColor},
      1px 0 ${textShadowColor}, -1px 0 ${textShadowColor},
      -2px 0 ${textShadowColor}, -3px 0 ${textShadowColor};
    touch-action: manipulation;
    transition: color 0.1s ease-out;

    @supports (text-decoration-skip-ink: auto) {
      background: none;
      text-decoration: underline;
      text-decoration-skip-ink: auto; /* stylelint-disable-line property-no-unknown */
      text-shadow: none;
    }

    &[href^='http'] {
      padding-right: 1em;

      &::after {
        content: '';
        height: 1em;
        position: absolute;
        width: 1em;
        background-position: center;
        background-repeat: no-repeat;
        background-size: 0.75em;
        ${EXTERNAL_LINK_ICON};
      }
    }
  }

  > h2 {
    font-size: ${MODULAR_SCALE_FACTOR ** 3}rem;
    line-height: ${() => H2_LINE_HEIGHT};
    ${BASE_FONT_BOLD};
  }

  > p {
    line-height: ${BASE_LINE_HEIGHT};
    margin: ${BASE_LINE_HEIGHT}em 0;
    word-break: break-word;

    @media (min-width: 950px) {
      line-height: ${BASE_LINE_HEIGHT * 1.1};
    }
  }

  ul {
    appearance: none !important;
    margin: 0 !important;

    li {
      line-height: ${BASE_LINE_HEIGHT * 1.2};
      margin-top: 0 !important;
    }
  }
`;

export default Article;
