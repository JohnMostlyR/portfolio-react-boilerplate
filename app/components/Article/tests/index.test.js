import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Article from '../index';

describe('<Article />', () => {
  it('Should render and match the snapshot', () => {
    const wrapper = shallow(<Article />);
    expect(wrapper).toMatchSnapshot();
  });
});
