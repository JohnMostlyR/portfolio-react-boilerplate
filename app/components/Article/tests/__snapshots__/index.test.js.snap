// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> Should render and match the snapshot 1`] = `
<StyledArticle
  loadingContent={false}
/>
`;
